lib/task-data/base-tasks/ucs-service-profile-discovery.js

```diff
@@ -0,0 +1,17 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = {
+    friendlyName: 'UCS Service Profile Discovery Job',
+    injectableName: 'Task.Base.UCS.Service.Profile.Discovery',
+    runJob: 'Job.Ucs.Service.Profile.Discovery',
+    requiredOptions: [
+        'uri'
+    ],
+    requiredProperties: {
+    },
+    properties: {
+        catalog: {}
+    }
+};
```

Comment:
This is pretty simple, just put a JSON schema in the property `optionsSchema`, this is very alike with the JSON schema in swagger ui.

This is an example: https://github.com/RackHD/on-tasks/blob/master/lib/task-data/base-tasks/provide-catalog-data.js